client/client.go

```diff
@@ -337,26 +347,28 @@ func (cli *Client) NegotiateAPIVersion(ctx context.Context) {
 // ([EnvOverrideAPIVersion]) environment variable, or if the client is initialized
 // with a fixed version ([WithVersion]), no negotiation is performed.
 //
-// If the API server's ping response does not contain an API version, we assume
-// we are connected with an old daemon without API version negotiation support,
-// and downgrade to the latest version of the API before version negotiation was
-// added (1.24).
+// If the API server's ping response does not contain an API version, it falls
+// back to the oldest API version supported.
 func (cli *Client) NegotiateAPIVersionPing(pingResponse types.Ping) {
```

Comment:
I'd like to get rid of this function; it's used in the CLI currently to use a "Ping" result that was triggered in some code, that uses the Ping for feature detection, then (likely only to avoid pinging "twice") reuses the ping result as optimization, but we probably shouldn't, and maybe either optimise the "ping" to only happen once for the lifecycle of the client, or have a method to query features.